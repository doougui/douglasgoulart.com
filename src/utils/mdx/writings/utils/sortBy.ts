import { Writing } from 'types/Writing';
import { parseToDate } from 'utils/time/parseToDate';
import { SortTypesKeys } from '../types';

function byNew(data: Writing[]) {
  return data.sort((a, b) => {
    const aAsDate = parseToDate(a.publishedAt);
    const bAsDate = parseToDate(b.publishedAt);
    return Number(bAsDate) - Number(aAsDate);
  });
}

function byName(data: Writing[]) {
  return data.sort((a, b) => (a.title > b.title ? 1 : -1));
}

export function sortBy(data: Writing[], type: SortTypesKeys): Writing[] {
  const sortTypes: { [key in SortTypesKeys]: () => Writing[] } = {
    new: () => byNew(data),
    name: () => byName(data),
  };

  return sortTypes[type]();
}
