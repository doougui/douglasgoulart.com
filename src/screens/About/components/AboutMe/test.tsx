import { render } from 'utils/tests';

import { AboutMe } from '.';

jest.mock('components/Image', () => ({
  __esModule: true,
  Image: () => <div data-testid="me-image" />,
}));

describe('<AboutMe />', () => {
  it('should render correctly', () => {
    const { container } = render(<AboutMe />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
