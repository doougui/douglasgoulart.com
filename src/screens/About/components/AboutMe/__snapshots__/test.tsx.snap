// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AboutMe /> should render correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 3rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 2rem;
}

.c2 {
  font-size: 1.6rem;
  line-height: 2.4rem;
  color: #EDEDEF;
}

@media (max-width:768px) {
  .c0 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

<section
  class="c0"
>
  <div
    data-testid="me-image"
  />
  <div
    class="c1"
  >
    <p
      class="c2"
    >
      I‘m a UI developer based in Barcelona. I‘m interested in design systems, web3, user/dev experience and under engineering.
    </p>
    <p
      class="c2"
    >
      I‘m currently doing UI/DX at Rainbow to help create the world‘s friendliest wallet.
    </p>
    <p
      class="c2"
    >
      Previously, I worked at Modulz. Where I co-created Radix, designed the Stitches API and led the developer community.
    </p>
  </div>
</section>
`;
